crates/wasi/src/preview1.rs

```diff
@@ -1290,12 +1290,8 @@ impl wasi_snapshot_preview1::WasiSnapshotPreview1 for WasiP1Ctx {
         let fd = self.get_file_fd(fd)?;
         self.as_wasi_impl()
             .advise(fd, offset, len, advice.into())
-            .await
-            .map_err(|e| {
-                e.try_into()
-                    .context("failed to call `advise`")
-                    .unwrap_or_else(types::Error::trap)
-            })
+            .await?;
+        Ok(())
```

Comment:
Was it intentional that the `.context` calls were all removed in this module?

Reply:
Intentional yeah, turns out that the `try_into` in these cases was infallible and couldn't ever fail. That meant that the error case never happened and the `.context(...)` calls were basically all dead code. I switched them all over to using `?` which leverages the `into` conversion automatically which is known to never fail.